### DISCUSSION TITLE
Vanilla `subscribe` does not trigger when a computed getter is updated

### DISCUSSION THREAD

**Discussion by sharathprabhal** [pmndrs/valtio#742] [Bug report] [answered]
I use Valtio as a store for in a project that uses both React and vanilla. I have a keyed object of items in a store. Based on user action, I set a "derived" currentObject and use a getter to select "current object" and perform operations on it. When the "current object" is updated, `useSnapshot` triggers but subscribe does not.

See repro code at https://codesandbox.io/s/valtio-subscribe-57p4st

Please let me know if this is a bug or if there is a alternate/better way to do this. Thanks!

**Comment by dai-shi** [comment]
It's how JS works:
```js
subscribe(store.currentObject, () => ...)

// ☝️ that means the following 👇 

const objectAtTheFirstTime = store.currentObject
subscribe(objectAtTheFirstTime, () => ...)
```

You should subscribe to the store: `subscribe(store, () => ...)`

**Comment by sharathprabhal** [comment]
thanks, @dai-shi . Do you have any recommendations for what I'm trying to achieve, which is essentially trying to listen to changes for a "selector" ? without having to re-subscribe every time the object changes.

**Reply by dai-shi** [reply, marked as answer]
```js
let lastObjectSnap
subscribe(store, () => {
  const currentObjectSnap = snapshot(store.currentObject);
  if (lastObjectSnap !== currentObjectSnap) {
    lastObjectSnap = currentObjectSnap
    // do your stuff
  }
})
```

**Reply by sharathprabhal** [reply]
Thank you!
